### PR TITLE
Build for linux/amd64 only

### LINKED ISSUE #226: Compile only for linux/amd64
As discussed in #219, there's little reason to support building for any other OS/arch. The defense for this assertion is that it makes almost* no sense to execute any component of the service catalog outside of a k8s cluster. If executing in a cluster, then we're assured of Linux as an OS and 99+% of the time, amd64 is a safe assumption for the architecture. (Sorry to folks running Kubernetes on Raspberry Pis-- you are few. 😉)

The benefit: A good deal of build process complexity can be eliminated if make targets like `controller-image` (for instance) can start assuming that any binaries that exist in the `bin` dir were compiled for linux/amd64. (Currently, we are forced to re-compile immediately before building an image because we have no assurances that the binaries in `bin` were compiled for linux/amd64. 😞 )

\*The edge case where someone might want to run our binaries outside of a k8s cluster (e.g. using their local machine) would be someone wishing to connect those components to a remote k8s cluster. In real life, I don't feel like that's such a compelling thing to do that we should even concern ourselves with it.

cc @duglin

### PR DESCRIPTION
**Pull request by krancour**
Fixes #226 

cc @duglin

### CODE REVIEW

**Comment by MHBauer** [comment]
I build and test the apiserver on darwin, is there a way to make this flaggable?

**Comment by krancour** [comment]
@MHBauer what's your motivation for doing so?

**Comment by MHBauer** [comment]
Habit, ease. I started that way, and it's worked out pretty well so far.

**Comment by krancour** [comment]
@MHBauer, `make test` is unaffected by this change, so zeroing in on `make build`...

If you currently build for darwin, what, if anything are you doing with the resulting binaries? You're executing them on your Mac, i.e. outside of any k8s cluster?

**Comment by MHBauer** [comment]
only the apiserver. As long as `go build` works after an initial codegen, I can adapt.

I have this aliased for personal use: `make apiserver && ./bin/apiserver --etcd-servers http://localhost:2379`. Then I switch tabs and curl/kubectl against it.

I suppose we should start having a more solid testing regime.

**Comment by MHBauer** [comment]
Let's put something in the readme that says we only support x64 and I'll be on board.

**Comment by krancour** [comment]
@MHBauer it's mentioned in the devguide. Sufficient?

https://github.com/kubernetes-incubator/service-catalog/pull/230/files#diff-12ea96d52d2082315f3bd85387bd452aR67

**Comment by MHBauer** [comment]
That's a good start. I mean more of a toplevel README type notification. As obvious as it seems, I feel it needs to be mentioned. Maybe I'm being overly sensitive.

**Comment by krancour** [comment]
@MHBauer I'll add something at the top level.

**Comment by MHBauer** [comment]
LGTM

File: Makefile [hunk 1]
```diff
@@ -35,16 +35,14 @@ NEWEST_GO_FILE = $(shell find $(SRC_DIRS) -name \*.go -exec $(STAT) {} \; \
                    | sort -r | head -n 1 | sed "s/.* //")
 TYPES_FILES    = $(shell find pkg/apis -name types.go)
 GO_VERSION     = 1.7.3
-GO_BUILD       = go build -i -v -ldflags "-X $(SC_PKG)/pkg.VERSION=$(VERSION)"
+GO_BUILD       = env GOOS=linux GOARCH=amd64 go build -i -v -ldflags "-X $(SC_PKG)/pkg.VERSION=$(VERSION)"
```
Comment (duglin) [inline Makefile]: do we need the "env" part?

Comment (krancour) [inline Makefile]: We do. If this is run in a container, Docker doesn't like when the command you pass it leads off with ENV=val. Using `env` solves that problem.

File: Makefile [hunk 2]
```diff
@@ -180,34 +170,26 @@ clean:
...
-	SC_GOOS=linux SC_GOARCH=amd64 BINDIR=bin/linux_amd64 $(DOCKER_FLAG) \
-	  $(MAKE) bin/linux_amd64/controller
-	cp pkg/controller/catalog/Dockerfile $(BINDIR)/linux_amd64/
-	docker build -t controller:$(VERSION) $(BINDIR)/linux_amd64
-	rm -f $(BINDIR)/linux_amd64/Dockerfile
+registry-image: contrib/registry/Dockerfile $(BINDIR)/registry
+	cp contrib/registry/Dockerfile $(BINDIR)
+	cp contrib/registry/data/charts/*.json $(BINDIR)
+	docker build -t registry:$(VERSION) $(BINDIR)
+	rm -f $(BINDIR)/*.json
```
Comment (duglin) [inline Makefile]: rm Dockerfile too?

Comment (krancour) [inline Makefile]: That was missing prior to this change, but I don't mind fixing that here if you don't mind me doing so.

Comment (krancour) [inline Makefile]: Fixed it.

File: Makefile (hunk 1 above)
Comment (duglin) [inline Makefile]: ah ok.
Can you wrap this at 80 cols?  :-)

**Comment by duglin** [comment]
I'd really like to get some google eyes on this since I think they're the other folks that might have strong opinions on this.

**Comment by pmorie** [comment]
LGTM to me personally.  I will say that k8s doesn't do an amazing job of maintaining support for mac development and I think it is wise to support only Linux for the moment.

**Comment by pmorie** [comment]
@bmelville @vaikas-google @martinmaly care to weigh in on this?

**Comment by arschles** [comment]
@krancour LGTM, merging

**Pull request merged**